### ISSUE TITLE
basepath and path issue

### ISSUE THREAD

**Issue by tljungberg** [domaindrivendev/Swashbuckle#1218]
Hi,

I have a bit of an issue that I am struggling to solve.

We have used swashbuckle 5 for some time, however now we are implementing API management in front of out API's and I need to essentially rewrite the base path.

The application is hosted in IIS in the root and the endpoints are https://{rooturl}/api/v40/method/function

I need to rewrite this so that the basePath becomes /api/v40 and the endpoints have that portion stripped to become just /method/function

In other words, currently:

basePath /
Path /api/v40/method/function

Desired:

basePath /api/v40
Path /method/function

Is there anyway to create some filter to perform this?

Thanks,

Tobey

**Comment by bragma** [comment]
I have the same need, I am not sure how to proceed

**Comment by pseudobjorn** [comment]
I have the same issue, but only when there are multiple versions involved. I have posted my question on Stack Overflow, and any help on this would be greatly appreciated.

https://stackoverflow.com/questions/49655190/how-can-i-populate-the-basepath-of-a-swagger-document-when-i-have-multiple-versi

@tljungberg and @bragma My question on Stack Overflow includes an example of how you can reach your goal, if your API only has a single version, using PreSerializeFilters.

**Comment by bragma** [comment]
@pseudobjorn yes I've solved using a IDocumentFilter:

		public class BasePathDocumentFilter : IDocumentFilter
		{
			public void Apply(SwaggerDocument swaggerDoc, DocumentFilterContext context)
			{
				string CommonPath(IEnumerable<string> strings)
				{
					var first = strings.FirstOrDefault();
					if (first == null)
					{
						return string.Empty;
					}

					var minimumLength = strings.Min(x => x.Length);
					var commonChars = Enumerable
						  .Range(0, minimumLength)
						  .Count(i => strings.All(y => y[i] == first[i]));

					var commonStart = first.Substring(0, commonChars);
					var lastSlash = commonStart.LastIndexOf("/");
					return lastSlash == -1 ? commonStart : commonStart.Remove(lastSlash);
				}

				var commonPath = CommonPath(swaggerDoc.Paths.Keys);
				if (commonPath.Length > 0)
				{
					swaggerDoc.BasePath = commonPath;
					swaggerDoc.Paths = swaggerDoc.Paths
						.Select(o => new KeyValuePair<string, PathItem>(o.Key.Substring(commonPath.Length), o.Value))
						.ToDictionary(n => n.Key, n => n.Value);
				}
			}
		}
